### PR TITLE
Cleanup user docs

### PR DESCRIPTION
**Pull request by standage**
Lots of out-of-date stuff here. Doing some general cleanup. Work in progress.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality, is it tested?
- [x] `make format diff_pylint_report doc` Is it well formatted?
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by standage** [comment]
Ready for review! @ctb @luizirber

File: doc/screed.rst [hunk 1]
```diff
@@ -1,181 +1,162 @@
...
+
+    $ screed db <fasta/fastq file>
+
+Dumping a database to a file
+----------------------------
+
+.. code::
+
+    $ screed dump_to_fasta <screed database file> <fasta output>
+    $ screed dump_to_fastq <screed database file> <fastq output>
```
Comment (luizirber) [inline doc/screed.rst]: need to drop the `_to_` here

File: doc/screed.rst [hunk 2]
```diff
@@ -1,181 +1,162 @@
...
+gzip-compressed, or bzip2-compressed. screed natively supports FASTA and FASTQ
+databases creation. If your sequences are in a different format see the
+developer documentation on :doc:`dev/parsers`.
 
-Creating a database from the API
---------------------------------
-
-From a Python prompt type::
+Creating a database
+-------------------
```
Comment (luizirber) [inline doc/screed.rst]: can also document `screed.make_db` here

Comment (standage) [inline doc/screed.rst]: Is that different from `screed db` above?

Comment (standage) [inline doc/screed.rst]: @luizirber Updated. Note, currently `import screed; screed.make_db(file)` doesn't work. It looks like you're addressing that in #63, but for now I have `import screed; screed.createscreed.make_db(file)` as that is what works.

Comment (standage) [inline doc/screed.rst]: Updated to `screed.make_db` now that #63 is merged.

**Comment by standage** [comment]
Ready for re-review! @luizirber

File: doc/screed.rst [hunk 3]
```diff
@@ -1,175 +1,166 @@
...
-uncompressed, gzipped, or bz2-zipped.
+Dumping a database to a file
+----------------------------
 
-Creating a database from the API
---------------------------------
+.. code::
 
-From a Python prompt type::
+    $ screed dump_fasta <screed database file> <fasta output>
```
Comment (ctb) [inline doc/screed.rst]: there's also a `dump_to_fastq` still in this file (tho not in the diff)

**Pull request merged**